### DISCUSSION TITLE
JSBSim featured in a scientific article on Nonlinear Dynamics (NODY, by Springer)

### DISCUSSION THREAD

**Discussion by agodemar** [JSBSim-Team/jsbsim#946] [Show and tell]
Hello developers and users of JSBSim. I'm delighted to share that my article

["A deep reinforcement learning control approach for high-performance aircraft"](https://link.springer.com/article/10.1007/s11071-023-08725-y)

went open access on the prestigious _Nonlinear Dynamics_, an International Journal of Nonlinear Dynamics and Chaos in Engineering Systems by Springer.

Many thanks to my co-authors Sabato Manfredi and Paolo Maria D'Onza.

#### Details

The article is available online here https://doi.org/10.1007/s11071-023-08725-y or as a PDF here https://link.springer.com/content/pdf/10.1007/s11071-023-08725-y.pdf.

The research features an application of Reinforcement Learning (RL) to flight control and guidance.
The work investigates the use of Deep Reinforcement Learning (DRL) to manage and control a full-blown 6-DoF nonlinear model of a military aircraft, with a Deep Deterministic Policy Gradient (DDPG) agent.

The chosen high-performance airplane is the F-16. The F-16 flight dynamics model is provided by the JSBSim flight simulation software library, exposed to the simulation environment via its Matlab/Simulink interface.

#### Tools

✈️ Matlab 👉 [Reinforcement Learning toolbox](https://www.mathworks.com/products/reinforcement-learning.html)

✈️ JSBSim 👉 [S-Function as interface to MATLAB/Simulink](https://github.com/JSBSim-Team/jsbsim/issues/491)

See also 👉 [F-16 model details](https://github.com/JSBSim-Team/jsbsim/tree/master/aircraft/f16)

#### Further reading

[For more details on this work, here is my series of LinkedIn posts on the same subject](https://www.linkedin.com/posts/agostino-de-marco-08398a7_matlab-reinforcement-learning-activity-6930166624477999104-LCLG).

**Comment by seanmcleod** [comment]
Read the paper already after you posted it on LinkedIn, might have a couple of questions 😉

**Comment by seanmcleod** [comment]
So after reading your paper and then coming across this recent paper - [Champion-level drone racing using deep reinforcement learning](https://www.nature.com/articles/s41586-023-06419-4) I think an interesting combination would be using RL to fly a Red Bull Air Race course. Comparing the maneuvers the RL agent comes up with to those of the Red Bull Air Race pilots, and comparing the performance differences.

Maybe starting off with something simpler, i.e. see whether a RL agent could match or better the climb to altitude record performance of the Streak Eagle.

![image](https://github.com/JSBSim-Team/jsbsim/assets/824856/b0a3e628-6c9b-4c42-82d0-89628279ffbd)

Would be interesting to understand the physics behind why this is the optimal climb profile to 20,000m.  

Here is a video showing all of the different climb to altitude record flights - [F-15 Streak Eagle Record Flights](https://www.youtube.com/watch?v=8_f8MOXh7qw). The record attempts for lower altitudes, e.g. 3,000m, 10,000m etc. used much simpler climb profiles.
